File: EnforcerWeb/src/app/model/LicensedArtifact.ts
export class LicensedArtifact {
	public licensed_artifact_id: number;
	public project_id: number;
	public artifact_id: number;
	public contract?: string;
	public vendor?: string;
	public approval_architect?: string;
	public approval_ts?: Date;

	constructor( obj: Object = {}) {
		Object.assign(this, obj) ;
	}
 
	public toJSON() {
		return {
			licensed_artifact_id: this.licensed_artifact_id,
			project_id: this.project_id,
			artifact_id: this.artifact_id,
			contract: this.contract,
			vendor: this.vendor,
			approval_architect: this.approval_architect,
			approval_ts: this.approval_ts,
		}
	}
}

File: EnforcerServices/src/main/java/com/webforged/enforcer/management/data/LicensedArtifact.java
package com.webforged.enforcer.management.data;

import java.time.Instant;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
@Table("licensedartifacts")
public class LicensedArtifact {
	
	@Id
	public Long lic_artifact_id ;
	private Long artifact_id;
	private Long project_id;
	private String contract;
	private String vendor;
	private String approval_architect;
	private Instant approval_ts;
	
	public LicensedArtifact() {
	}
}